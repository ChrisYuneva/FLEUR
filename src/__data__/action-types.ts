export const GOODS = {
    REQUEST: 'GOODS_REQUEST',
    SUCCESS: 'GOODS_SUCCESS',
    FAILURE: 'GOODS_FAILURE',
    SET_GENDER: 'GOODS_SET_GENDER',
    SET_TYPE: 'GOODS_SET_TYPE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};
