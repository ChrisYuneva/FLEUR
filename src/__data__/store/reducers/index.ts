import { combineReducers } from 'redux';
import goodsReducer, { GoodsState } from './goods';
import basketReducer, { BasketState } from './basket';

export type AppStore = {
    goods: GoodsState;
    basket: BasketState;
};

export default combineReducers<AppStore>({
    goods: goodsReducer,
    basket: basketReducer,
});
