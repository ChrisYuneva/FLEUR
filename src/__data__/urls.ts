import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('repos');

export const baseUrl = navigations['repos'];
export const URLs = {
    basket: {
        url: navigations['link.repos.basket'],
    },
    home: {
        url: navigations['link.repos.home'],
    },
    home_man: {
        url: navigations['link.repos.home_man'],
    },
    showcase_men: {
        url: navigations['link.repos.showcase_men'],
    },
    showcase_women: {
        url: navigations['link.repos.showcase_women'],
    },
    ordering: {
        url: navigations['link.repos.ordering'],
    },
};
