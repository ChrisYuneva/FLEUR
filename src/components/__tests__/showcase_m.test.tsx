import React from 'react';
import { mount } from 'enzyme';
import { describe, it, expect } from '@jest/globals';
import { ShowcaseMan } from '../index';
import { store } from '../../__data__/store';
import { Provider } from 'react-redux';

describe('<ShowcaseMan />', () => {
    const title = 'title';
    const caption = 'caption';

    it('Отрисовывается без ошибок', () => {
        const wrapper = mount(
            <Provider store={store}>
                <ShowcaseMan title={title} caption={caption} />
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
