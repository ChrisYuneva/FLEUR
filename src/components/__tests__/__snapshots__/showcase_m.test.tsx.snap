// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowcaseMan /> Отрисовывается без ошибок 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowcaseMen)
    caption="caption"
    title="title"
  >
    <ShowcaseMen
      caption="caption"
      setGender={[Function]}
      setType={[Function]}
      title="title"
    >
      <div
        className="wrapper"
      >
        <nav
          className="nav"
        >
          <Link
            className="name"
            to="/home"
          >
            <a
              className="name"
              href="/home"
            >
              FLEUR
            </a>
          </Link>
          <Link
            className="link"
            to="/showcase_jackets_men"
          >
            <a
              className="link"
              href="/showcase_jackets_men"
            >
              Куртки
            </a>
          </Link>
          <Link
            className="link"
            to="/showcase_shirts_men"
          >
            <a
              className="link"
              href="/showcase_shirts_men"
            >
              Рубашки
            </a>
          </Link>
          <Link
            className="link"
            to="/showcase_hoodies_men"
          >
            <a
              className="link"
              href="/showcase_hoodies_men"
            >
              Толстовки
            </a>
          </Link>
          <Link
            className="link"
            to="/showcase_tshirts_men"
          >
            <a
              className="link"
              href="/showcase_tshirts_men"
            >
              Футболки
            </a>
          </Link>
          <Link
            className="link"
            to="/showcase_jeans_men"
          >
            <a
              className="link"
              href="/showcase_jeans_men"
            >
              Джинсы
            </a>
          </Link>
          <Link
            className="link"
            onClick={[Function]}
            to="/showcase_dress"
          >
            <a
              className="link"
              href="/showcase_dress"
              onClick={[Function]}
            >
              Женщины
            </a>
          </Link>
        </nav>
        <div
          className="div"
        >
          <header
            className="header"
          >
            <Link
              className="ref"
              to="/basket"
            >
              <a
                className="ref"
                href="/basket"
              >
                <img
                  className="box"
                  src="FILE_MOCK"
                />
                <a
                  className="text"
                >
                  Корзина
                </a>
              </a>
            </Link>
          </header>
          <div
            className="scan"
          >
            <h2>
              title
            </h2>
            <span
              className="span"
            >
              caption
            </span>
            <div
              className="show"
            >
              <div
                className="showrow"
              />
            </div>
          </div>
        </div>
      </div>
    </ShowcaseMen>
  </Connect(ShowcaseMen)>
</Provider>
`;
